src/boot/efi/boot.c

```diff
@@ -2081,7 +2081,7 @@ static VOID config_load_xbootldr(
                                 continue;
 
                         /* Now load the GPT entry table */
-                        sz = ((h->SizeOfPartitionEntry * h->NumberOfPartitionEntries + 511) / 512) * 512;
+                        sz = ALIGN_TO(h->SizeOfPartitionEntry * h->NumberOfPartitionEntries, 512);
```

Comment:
hmm, these two values are 32bit, so you calc the product as 32bit too, and only then convert to 64bit through ALIGN_TO...

To be fully correct we should probably do something like this:

```
if (h->SizeOfPartitionEntry > UINTN_MAX / h->NumberOfPartitionEntries) /* overflow check */
        continue;

sz = ALIGN_TO((UINTN) h->SizeOfPartitionEntry * (UINTN) h->NumberOfPartitionEntries);
```